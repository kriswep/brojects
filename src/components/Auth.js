import React from 'react';
import { Button, Textfield } from 'react-mdl';

const Auth = ({ onSubmit, onChange, onClick, authtoken }) => (
  <section>
    <form onSubmit={onSubmit}>
      <Textfield
        label="Please provide your auth token"
        id="authtoken"
        onChange={onChange}
        value={authtoken}
        />
    </form>
    <Button raised colored ripple
      onClick={onClick}
      >Get Stuff
    </Button>
  </section>
)

export default Auth;
