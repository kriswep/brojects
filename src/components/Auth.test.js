import React from 'react';
import ReactDOM from 'react-dom';
// import ReactTestRenderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import 'react-mdl/extra/material.css';
import 'react-mdl/extra/css/material.indigo-pink.min.css';
import 'react-mdl/extra/material.js';

import Auth from './Auth';

const noop = () => { };

describe('Auth', () => {

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Auth />, div);
  });

  it('renders as expected', () => {
    // const renderer = ReactTestRenderer.create(<Auth />);
    const wrapper = shallow(<Auth
      onChange={noop}
      onSubmit={noop}
      onClick={noop}
      authtoken={'authtoken'}
    />);
    // expect(wrapper.find('header').hasClass('App-header')).toBe(true);

    expect(toJson(wrapper)).toMatchSnapshot();
    // expect(renderer.toJSON()).toMatchSnapshot();
  });

  // it('renders a Header', () => {
  //   const wrapper = shallow(<Auth />);
  // expect(wrapper.toMatchSnapshot)
  //   // expect(wrapper.find('header').hasClass('App-header')).toBe(true);
  // });
});
