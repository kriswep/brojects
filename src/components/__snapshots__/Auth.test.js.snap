exports[`Auth renders as expected 1`] = `
<section>
  <form
    onSubmit={[Function]}>
    <Textfield
      id="authtoken"
      label="Please provide your auth token"
      onChange={[Function]}
      value="authtoken" />
  </form>
  <Button
    colored={true}
    onClick={[Function]}
    raised={true}
    ripple={true}>
    Get Stuff
  </Button>
</section>
`;
